src/NHibernate/Multi/LinqBatchItem.cs

```diff
@@ -0,0 +1,110 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using NHibernate.Linq;
+using NHibernate.Util;
+using Remotion.Linq.Parsing.ExpressionVisitors;
+
+namespace NHibernate
+{
+	public partial class LinqBatchItem<T> : QueryBatchItem<T>
+	{
+		private static Delegate _postExecuteTransformer;
+		private static NhLinqExpression _linqEx;
```

Comment:
These members are specific to the query and cannot be static. They must be instance variables.

Reply:
ouch..